### PR TITLE
feat(extract-subsystems): extract `Context`, `@In`, `InjectionHelper` and `@Share` to Subsystem

### PR DESCRIPTION
**Pull request by DarkWeird**
### Contains

Part of #4304.

Provide possible to Injectable classes(`@Share`, `@In` and `Context` usage) with another subsystems.

Should be replaced with gestalt v8 (di)

### How to test

1.Run your favorite game (hint: Terasology)
2. Check it don't fail whit null pointer exception at `@In` field.

### Outstanding before merging

### CODE REVIEW

**Comment by DarkWeird** [comment]
[`Context` Subsystem extraction](https://trello.com/c/lyIYImDl/194-context-subsystem-extraction)

**Review by keturn — CHANGES_REQUESTED** [review]
CI did not run any `org.terasology.engine.registry` tests: https://jenkins.terasology.io/teraorg/job/Terasology/job/engine/job/PR-4930/lastCompletedBuild/testReport/

**Comment by DarkWeird** [comment]
@keturn 
> CI did not run any `org.terasology.engine.registry` tests: https://jenkins.terasology.io/teraorg/job/Terasology/job/engine/job/PR-4930/lastCompletedBuild/testReport/

Yeah.
Our pipeline ignores `test` task.
test-related tasks(in run order):
1.  `gradle unitTest`
2.  `gradle check -x test` (-x `test` - excludes `test` task)
3.  `gradle integrationTest`

This affects all `:subsystems:*` :(

File: subsystems/Context/src/test/java/org/terasology/engine/registry/CoreRegistryTest.java [hunk 1]
```diff
@@ -31,7 +26,7 @@ public void setup() {
     @Test
     public void testContextChange() {
         CoreRegistry.setContext(new ContextImplementation());
-        assertNotEquals(CoreRegistry.get(Context.class), context);
+        Assertions.assertNotEquals(CoreRegistry.get(Context.class), context);
```
[the remark this answers is not in the document]
Comment (DarkWeird) [reply subsystems/Context/src/test/java/org/terasology/engine/registry/CoreRegistryTest.java:29]: Just personal prefer to avoid static imports

**Comment by DarkWeird** [comment]
about `test` task - not this PR issue

**Comment by jdrueckert** [comment]
@DarkWeird In today's contributor meeting the question came up, whether it makes sense to invest into this when the "proper" way to do this seems to be gestalt v8...?
fyi @keturn

**Comment by DarkWeird** [comment]
> @DarkWeird In today's contributor meeting the question came up, whether it makes sense to invest into this when the "proper" way to do this seems to be gestalt v8...?
yeah :D gestalt v8 is "proper" way

see description:
>Should be replaced with gestalt v8 (di)

this pr is trying to split injections before gestalt v8 usage

**Comment by jdrueckert** [comment]
@DarkWeird Hehe okay, let me rephrase my question: What's the benefit ofextracting these things into a subsystem before using gestalt v8?

**Comment by DarkWeird** [comment]
> @DarkWeird Hehe okay, let me rephrase my question: What's the benefit ofextracting these things into a subsystem before using gestalt v8?

Path to extract another things (to subsystems) before then gestalt v8 will be integrated.
This submodule easy to remove when gestaltv8 di will be integrated

**Comment by jdrueckert** [comment]
From my pov, if this makes it easier to get rid of it after migrating to gestalt v8, I'm fine with reviewing and merging it (provided you resolve the conflicts first 😅 )
However, IIRC from our last meeting, keturn had some concerns about the required effort to make this a subsystem first. @keturn can you comment on the details of your concerns some more before we move on with this?
